vendor/k8s.io/kubernetes/pkg/kubeapiserver/authenticator/config.go

```diff
@@ -189,14 +189,15 @@ func (config Config) New() (authenticator.Request, *spec.SecurityDefinitions, er
 		}
 
 		tokenAuthenticators = append(tokenAuthenticators, webhookTokenAuth)
+	} else {
+		// openshift gets injected here as a separate token authenticator because we also add a special group to our oauth authorized
+		// tokens that allows us to recognize human users differently than machine users.  The openAPI is always correct because we always
+		// configuration service account tokens, so we just have to create and add another authenticator.
+		// TODO make this a webhook authenticator and remove this patch.
```

Comment:
please open a BZ

Reply:
Added enhancement PR link instead

Reply:
This is part of our move to webhooks in general as described in the enhancement. No need to track that in BZ.